### PR TITLE
Enable symbol upload on Publish

### PR DESCRIPTION
**Pull request by dannyvv**
###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7460)

### CODE REVIEW

File: .ado/publish.yml [hunk 1]
```diff
@@ -282,12 +286,13 @@ jobs:
           scanType: 'Register'
           failOnAlert: $(FailCGOnAlert) # Alerts must fail the run to prevent codesigning
 
-    # Disable for now, not sure this works on github projects anyway.
-    # - task: PublishSymbols@2
-    #   displayName: 'Publish symbols path'
-    #   inputs:
-    #     SearchPattern: 'vnext/target/**/*.pdb'
-    #     SymbolServerType: TeamServices
+      # Make symbols available through http://symweb.
```
Comment (JunielKatarn) [inline .ado/publish.yml:289]: What has changed?

Are GitHub projects now supported?

Comment (dannyvv) [reply .ado/publish.yml:289]: We are indexing the pdb's with sourcelink. So we just have to Publish, not Index and Publish... 
Now that our ci runs on microsoft.visualstudio.com this should automatically chain them to http://symweb for internal msft usage.

If this is succesfull we can consider removing the pdb's from nuget.

Comment (acoates-ms) [reply .ado/publish.yml:289]: Does this help external users any?  Will external users be able to get the symbols/source information from the nuget built dll?

Comment (dannyvv) [reply .ado/publish.yml:289]: This change not yet, but per: https://www.1eswiki.com/wiki/Using_Azure_DevOps_Symbols#Background this should be feasable.

Comment (dannyvv) [reply .ado/publish.yml:289]: we'll have two options: 
* https://msdl.microsoft.com/download/symbols
* https://ms.artifacts.visualstudio.com/defaultcollection/_apis/symbol/symsrv 

I'll try for the first one after this works for internal users.

Comment (asklar) [reply .ado/publish.yml:289]: main thing is to unblock internal users, we can fix it for the community separately/later

Comment (dannyvv) [reply .ado/publish.yml:289]: @JunielKatarn /@asklar / @NickGerleman : Friendly ping to see if we agree to test this first and then look at public.

Comment (JunielKatarn) [reply .ado/publish.yml:289]: I agree with @asklar.
Let's take one step at a time.

Comment (dannyvv) [reply .ado/publish.yml:289]: @JunielKatarn / @asklar: Would either of you mind signing off on the PR then :D

**Review by NickGerleman — APPROVED** [review]

**Review by asklar — APPROVED** [review]

**Pull request merged**